src/hooks/useStopPatchPolling.ts

```diff
@@ -0,0 +1,25 @@
+import React, { useEffect } from "react";
+import { PatchStatus } from "types/patch";
+
+interface Props {
+  hasData: boolean;
+  status: string;
+  activated: boolean;
+  stopPolling: () => void;
+}
+
+export const useStopPatchPolling = ({
+  hasData,
+  status,
+  activated,
+  stopPolling,
+}: Props) => {
+  if (
+    hasData &&
+    (status === PatchStatus.Failed ||
```

Comment:
I realize it was like this before but the data can still change if one of these conditions is met. Is the intent that we want to stop polling if we're absolutely sure the data won't change anymore?

Reply:
right now the live updating is not sophisticated but i think that has to do with the behavior/UX of the entire app not being fully fleshed out yet. also, if a patch changes from `not running` to `running again` then an event occurred that will most likely trigger a refetch, such as restarting the query.

Reply:
Right, I was referring to the scenario when someone else makes changes to the patch you're looking at. Would it be so bad to let it poll constantly? Or perhaps we can stop it when the tab loses focus and restart it when it gains focus

Reply:
That's right. We want to stop polling if we are sure the data won't change. Do you know how we can adjust the condition to achieve this?

Reply:
It can always change since you can restart patches

Reply:
let's increase poll - what do y'all think? 5 seconds? 10 seconds? And then we can let it poll constantly and use `useEffect(() => stopPolling, [stopPolling])` to ensure polling stops when component unmounts.

Reply:
I removed the useStopPatchPolling hook and applied the useEffects in this commit: 6c65a23. I think 5 seconds is reasonable.